effectful-th/effectful-th.cabal

```diff
@@ -0,0 +1,89 @@
+cabal-version:      2.4
+build-type:         Simple
+name:               effectful-th
+version:            0.1
+license:            BSD-3-Clause
+license-file:       LICENSE
+category:           Control
+maintainer:         andrzej@rybczak.net
+author:             Andrzej Rybczak
+synopsis:           Template Haskell utilities for the effectful library.
+
+description: This package provides several Template Haskell helper functions to
+             aid the development of effects for the 
+             @<https://hackage.haskell.org/package/effectful effectful>@
+             library.
+
+extra-source-files: CHANGELOG.md
+
+tested-with: GHC ==8.8.4 || ==8.10.7 || ==9.0.2 || ==9.2.1
+
+bug-reports:   https://github.com/arybczak/effectful/issues
+source-repository head
+  type:     git
+  location: https://github.com/arybczak/effectful.git
+
+common language
+    ghc-options:        -Wall -Wcompat
+
+    default-language:   Haskell2010
+
+    default-extensions: BangPatterns
+                        ConstraintKinds
+                        DataKinds
+                        DeriveFunctor
+                        DeriveGeneric
+                        FlexibleContexts
+                        FlexibleInstances
+                        GADTs
+                        GeneralizedNewtypeDeriving
+                        LambdaCase
+                        MultiParamTypeClasses
+                        NoStarIsType
+                        RankNTypes
+                        RecordWildCards
+                        RoleAnnotations
+                        ScopedTypeVariables
+                        StandaloneDeriving
+                        TupleSections
+                        TypeApplications
+                        TypeFamilies
+                        TypeOperators
+
+library
+    import:         language
+
+    ghc-options:    -O2
+
+    build-depends:    base                >= 4.13      && < 5
+                    , containers
+                    , effectful-core
+                    , template-haskell
```

Comment:
Need to provide upper and lower version bound, it's very likely something breaks in GHC 9.4.